pkg/apis/apps/types.go

```diff
@@ -44,6 +44,21 @@ type StatefulSet struct {
 	Status StatefulSetStatus
 }
 
+// PodCreationPolicyType defines the policy for creating pods under a stateful set.
+type PodCreationPolicyType string
+
+const (
+	// OrderedReadyPodCreation will create pods in strictly increasing order on
+	// scale up and strictly decreasing order on scale down, progressing only when
+	// the previous pod is ready or terminated. At most one pod will be changed
+	// at any time.
+	OrderedReadyPodCreation PodCreationPolicyType = "OrderedReady"
+	// ParallelPodCreation will create and delete pods as soon as the stateful set
```

Comment:
It's not about scaling, since replacement is also covered.  PodManagementPolicy seems vaguer - maybe too vague?  A new user seeing "PodCreationPolicy" is unlikely to be confused about what it does.  Other words that a brand new user to Kubernetes would be unlikely to be confused about?

> This is the Policy that controls how Pods are <created/updated/deleted/managed/replaced/instantiated>